compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2723,7 +2760,8 @@ public void visit(BLangExpressionStmt exprStmtNode, AnalyzerData data) {
         BType bType = typeChecker.checkExpr(expr, stmtEnv, data.prevEnvs, data.commonAnalyzerData);
         if (bType != symTable.nilType && bType != symTable.semanticError &&
                 expr.getKind() != NodeKind.FAIL &&
-                !types.isNeverTypeOrStructureTypeWithARequiredNeverMember(bType)) {
+                !types.isNeverTypeOrStructureTypeWithARequiredNeverMember(bType) &&
+                !isUnionWithSingleNilTypeMember(bType) && !isFiniteWithNilMember(bType)) {
```

Comment:
I would extract the type check for nil type here to a separate method (which in turn will call the other methods). E.g., `isSubTypeOfNil`

Reply:
Also, can't we use isAssignable with symTable.nilType instead of these individual checks?

Reply:
> Also, can't we use isAssignable with symTable.nilType instead of these individual checks?

@suleka96, any update on this?